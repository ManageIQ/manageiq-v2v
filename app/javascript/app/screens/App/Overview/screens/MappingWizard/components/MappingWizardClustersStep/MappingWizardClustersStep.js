import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'patternfly-react';

class MappingWizardClustersStep extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedTargetCluster: null, // eslint-disable-line react/no-unused-state
      selectedSourceClusters: [], // eslint-disable-line react/no-unused-state
      mappings: [], // eslint-disable-line react/no-unused-state
      selectedMapping: null // eslint-disable-line react/no-unused-state
    };
  }

  componentDidMount() {
    const {
      fetchSourceClustersUrl,
      fetchSourceClustersAction,
      fetchTargetClustersUrl,
      fetchTargetClustersAction
    } = this.props;

    fetchSourceClustersAction(fetchSourceClustersUrl);
    fetchTargetClustersAction(fetchTargetClustersUrl);
  }

  render() {
    const {
      isFetchingSourceClusters,
      sourceClusters,
      isFetchingTargetClusters,
      targetClusters
    } = this.props;

    if (!isFetchingSourceClusters && !isFetchingTargetClusters) {
      return (
        <div>
          <p>Source Clusters</p>
          {sourceClusters.map(cluster => (
            <div key={cluster.id}>
              <span>
                <b>Name</b>: {cluster.name} &nbsp;
              </span>
              <span>
                <b>ID</b>: {cluster.id} &nbsp;
              </span>
            </div>
          ))}
          <br />
          <p>Target Clusters</p>
          {targetClusters.map(cluster => (
            <div key={cluster.id}>
              <span>
                <b>Name</b>: {cluster.name} &nbsp;
              </span>
              <span>
                <b>ID</b>: {cluster.id} &nbsp;
              </span>
            </div>
          ))}
        </div>
      );
    }
    return null;
  }
}

MappingWizardClustersStep.propTypes = {
  fetchSourceClustersUrl: PropTypes.string,
  fetchSourceClustersAction: PropTypes.func,
  fetchTargetClustersUrl: PropTypes.string,
  fetchTargetClustersAction: PropTypes.func,
  sourceClusters: PropTypes.arrayOf(PropTypes.object),
  targetClusters: PropTypes.arrayOf(PropTypes.object),
  isFetchingSourceClusters: PropTypes.bool,
  isFetchingTargetClusters: PropTypes.bool
};
MappingWizardClustersStep.defaultProps = {
  fetchSourceClustersUrl: '',
  fetchSourceClustersAction: noop,
  fetchTargetClustersUrl: '',
  fetchTargetClustersAction: noop,
  isFetchingSourceClusters: true,
  isFetchingTargetClusters: true
};

export default MappingWizardClustersStep;
