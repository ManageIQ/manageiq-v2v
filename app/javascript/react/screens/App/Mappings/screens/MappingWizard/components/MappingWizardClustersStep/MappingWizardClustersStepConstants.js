import { OPENSTACK, RHV } from '../../MappingWizardConstants';

export const FETCH_V2V_SOURCE_CLUSTERS = 'FETCH_V2V_SOURCE_CLUSTERS';
export const FETCH_V2V_TARGET_CLUSTERS = 'FETCH_V2V_TARGET_CLUSTERS';
export const QUERY_V2V_PROVIDERS = 'QUERY_V2V_PROVIDERS';

export const CONVERSION_HOST_TYPES = {
  [RHV]: 'ManageIQ::Providers::Redhat::InfraManager::Host',
  [OPENSTACK]: 'ManageIQ::Providers::Openstack::CloudManager::Vm'
};

export const FETCH_TARGET_COMPUTE_URLS = {
  [RHV]:
    '/api/clusters?expand=resources' +
    '&attributes=ext_management_system.emstype,v_parent_datacenter,ext_management_system.name,hosts' +
    '&filter[]=ext_management_system.emstype=rhevm',
  [OPENSTACK]:
    '/api/cloud_tenants?expand=resources&filter[]=ext_management_system.type=ManageIQ::Providers::Openstack::CloudManager&attributes=ext_management_system.name,ext_management_system.id'
};

export const QUERY_PROVIDERS_URL = '/api/providers';

export const TARGET_WARNING_MESSAGES = {
  [RHV]: __('You must enable at least one conversion host in the cluster. You can continue to create an infrastructure mapping that includes the target cluster, but you must enable a conversion host before running the migration plan.'), // prettier-ignore
  [OPENSTACK]: __('You must enable an RSA key pair on the OpenStack provider. You can continue to create an infrastructure mapping without the key pair, but you must enable it before running the migration plan.') // prettier-ignore
};
