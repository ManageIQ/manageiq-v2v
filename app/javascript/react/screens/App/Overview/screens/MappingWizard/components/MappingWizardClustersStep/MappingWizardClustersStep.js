import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'patternfly-react';

import DualPaneMapper from '../DualPaneMapper/DualPaneMapper';
import DualPaneMapperList from '../DualPaneMapper/DualPaneMapperList';
import DualPaneMapperCount from '../DualPaneMapper/DualPaneMapperCount';
import DualPaneMapperListItem from '../DualPaneMapper/DualPaneMapperListItem';
import ClustersStepTreeView from './components/ClustersStepTreeView';

class MappingWizardClustersStep extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedTargetCluster: null, // eslint-disable-line react/no-unused-state
      selectedSourceClusters: [], // eslint-disable-line react/no-unused-state
      mappings: [], // eslint-disable-line react/no-unused-state
      selectedMapping: null // eslint-disable-line react/no-unused-state
    };

    this.selectSourceCluster = this.selectSourceCluster.bind(this);
    this.selectTargetCluster = this.selectTargetCluster.bind(this);
    this.addMapping = this.addMapping.bind(this);
    this.selectMapping = this.selectMapping.bind(this);
    this.removeMapping = this.removeMapping.bind(this);
    this.removeAll = this.removeAll.bind(this);
  }

  componentDidMount() {
    const {
      fetchSourceClustersUrl,
      fetchSourceClustersAction,
      fetchTargetClustersUrl,
      fetchTargetClustersAction
    } = this.props;

    fetchSourceClustersAction(fetchSourceClustersUrl);
    fetchTargetClustersAction(fetchTargetClustersUrl);
  }

  selectSourceCluster(sourceCluster) {
    this.setState(prevState => {
      const isAlreadySelected = prevState.selectedSourceClusters.some(
        cluster => cluster.id === sourceCluster.id
      );
      if (isAlreadySelected) {
        return {
          selectedSourceClusters: prevState.selectedSourceClusters.filter(
            cluster => cluster.id !== sourceCluster.id
          )
        };
      }
      return {
        selectedSourceClusters: [
          ...prevState.selectedSourceClusters,
          sourceCluster
        ]
      };
    });
  }

  selectTargetCluster(targetCluster) {
    this.setState(() => ({ selectedTargetCluster: targetCluster }));
  }

  addMapping() {
    const { removeTargetCluster } = this.props;
    const { selectedTargetCluster } = this.state;
    removeTargetCluster(selectedTargetCluster);
    this.setState(prevState => ({
      selectedTargetCluster: null,
      selectedSourceClusters: [],
      mappings: [
        ...prevState.mappings,
        {
          ...prevState.selectedTargetCluster,
          text: prevState.selectedTargetCluster.name,
          state: {
            expanded: true
          },
          selectable: true,
          selected: false,
          nodes: prevState.selectedSourceClusters.map(cluster => ({
            ...cluster,
            text: cluster.name,
            icon: 'fa fa-file-o'
          }))
        }
      ]
    }));
  }

  selectMapping(selectedMapping) {
    this.setState(prevState => ({
      mappings: prevState.mappings.map(mapping => {
        if (mapping.id === selectedMapping.id) {
          return { ...mapping, selected: !mapping.selected };
        } else if (mapping.id !== selectedMapping.id && mapping.selected) {
          return { ...mapping, selected: false };
        }
        return mapping;
      }),
      selectedMapping
    }));
  }

  removeMapping() {
    this.setState(prevState =>
      // const { nodes, ...targetCluster } = prevState.selectedMapping;
      ({
        mappings: prevState.mappings.filter(
          mapping => !(mapping.id === prevState.selectedMapping.id)
        ),
        selectedMapping: null
        // sourceClusters: {
        //   ...prevState.sourceClusters,
        //   resources: [...prevState.sourceClusters.resources, ...nodes]
        // },
        // targetClusters: {
        //   ...prevState.targetClusters,
        //   resources: [...prevState.targetClusters.resources, targetCluster]
        // }
      })
    );
  }

  removeAll() {
    this.setState(prevState =>
      // const sourceClusters = { ...prevState.sourceClusters };
      // const targetClusters = { ...prevState.targetClusters };
      // prevState.mappings.forEach(mapping => {
      //   const { nodes, ...targetCluster } = mapping;
      //   sourceClusters.resources = [...sourceClusters.resources, ...nodes];
      //   targetClusters.resources = [...targetClusters.resources, targetCluster];
      // });
      ({
        mappings: []
        // sourceClusters,
        // targetClusters
      })
    );
  }

  render() {
    const {
      isFetchingSourceClusters,
      sourceClusters,
      isFetchingTargetClusters,
      targetClusters
    } = this.props;

    const {
      selectedTargetCluster,
      selectedSourceClusters,
      mappings,
      selectedMapping
    } = this.state;

    if (!isFetchingSourceClusters && !isFetchingTargetClusters) {
      return (
        <div className="mapping-wizard-clusters-step">
          <DualPaneMapper
            handleButtonClick={this.addMapping}
            validMapping={
              !(
                selectedTargetCluster &&
                (selectedSourceClusters && selectedSourceClusters.length > 0)
              )
            }
          >
            {sourceClusters && (
              <DualPaneMapperList listTitle="fix me">
                {sourceClusters.map(item => (
                  <DualPaneMapperListItem
                    item={item}
                    key={item.id}
                    selected={
                      selectedSourceClusters &&
                      selectedSourceClusters.some(
                        sourceCluster => sourceCluster.id === item.id
                      )
                    }
                    handleClick={this.selectSourceCluster}
                    handleKeyPress={this.selectSourceCluster}
                  />
                ))}
                <DualPaneMapperCount
                  selectedItems={selectedSourceClusters.length}
                  totalItems={sourceClusters.length}
                />
              </DualPaneMapperList>
            )}
            {targetClusters && (
              <DualPaneMapperList listTitle="fix me">
                {targetClusters.map(item => (
                  <DualPaneMapperListItem
                    item={item}
                    key={item.id}
                    selected={
                      selectedTargetCluster &&
                      selectedTargetCluster.id === item.id
                    }
                    handleClick={this.selectTargetCluster}
                    handleKeyPress={this.selectTargetCluster}
                  />
                ))}
              </DualPaneMapperList>
            )}
          </DualPaneMapper>
          <ClustersStepTreeView
            mappings={mappings}
            selectMapping={this.selectMapping}
            removeMapping={this.removeMapping}
            removeAll={this.removeAll}
            selectedMapping={selectedMapping}
          />
        </div>
      );
    }
    return null;
  }
}

MappingWizardClustersStep.propTypes = {
  fetchSourceClustersUrl: PropTypes.string,
  fetchSourceClustersAction: PropTypes.func,
  fetchTargetClustersUrl: PropTypes.string,
  fetchTargetClustersAction: PropTypes.func,
  sourceClusters: PropTypes.arrayOf(PropTypes.object),
  targetClusters: PropTypes.arrayOf(PropTypes.object),
  isFetchingSourceClusters: PropTypes.bool,
  isFetchingTargetClusters: PropTypes.bool,
  removeTargetCluster: PropTypes.func
};
MappingWizardClustersStep.defaultProps = {
  fetchSourceClustersUrl: '',
  fetchSourceClustersAction: noop,
  fetchTargetClustersUrl: '',
  fetchTargetClustersAction: noop,
  isFetchingSourceClusters: true,
  isFetchingTargetClusters: true,
  removeTargetCluster: noop
};

export default MappingWizardClustersStep;
