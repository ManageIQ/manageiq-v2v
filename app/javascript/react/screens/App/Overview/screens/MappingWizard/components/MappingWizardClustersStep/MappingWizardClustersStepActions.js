import URI from 'urijs';
import API from '../../../../../../../../common/API';
import {
  FETCH_V2V_SOURCE_CLUSTERS,
  FETCH_V2V_TARGET_CLUSTERS,
  REMOVE_TARGET_CLUSTER,
  REMOVE_SOURCE_CLUSTERS
} from './MappingWizardClustersStepConstants';

import {
  requestSourceClustersData,
  requestTargetClustersData
} from './mappingWizardClustersStep.fixtures';

const _getSourceClustersActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_SOURCE_CLUSTERS,
    payload: API.get(url)
  }).catch(error => {
    // redux-promise-middleware will automatically send:
    // FETCH_V2V_SOURCE_CLUSTERS_PENDING, FETCH_V2V_SOURCE_CLUSTERS_FULFILLED, FETCH_V2V_SOURCE_CLUSTERS_REJECTED

    // to enable UI development without the backend ready, i'm catching the error
    // and passing some mock data thru the FULFILLED action after the REJECTED action is finished.
    dispatch({
      type: `${FETCH_V2V_SOURCE_CLUSTERS}_FULFILLED`,
      payload: requestSourceClustersData.response
    });
  });

export const fetchSourceClustersAction = url => {
  const uri = new URI(url);
  return _getSourceClustersActionCreator(uri.toString());
};

const _getTargetClustersActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_TARGET_CLUSTERS,
    payload: API.get(url)
  }).catch(error => {
    // redux-promise-middleware will automatically send:
    // FETCH_V2V_TARGET_CLUSTERS_PENDING, FETCH_V2V_TARGET_CLUSTERS_FULFILLED, FETCH_V2V_TARGET_CLUSTERS_REJECTED

    // to enable UI development without the backend ready, i'm catching the error
    // and passing some mock data thru the FULFILLED action after the REJECTED action is finished.
    dispatch({
      type: `${FETCH_V2V_TARGET_CLUSTERS}_FULFILLED`,
      payload: requestTargetClustersData.response
    });
  });

export const fetchTargetClustersAction = url => {
  const uri = new URI(url);
  return _getTargetClustersActionCreator(uri.toString());
};

export const removeTargetCluster = targetClusterToRemove => dispatch =>
  dispatch({
    type: REMOVE_TARGET_CLUSTER,
    targetClusterToRemove
  });

export const removeSourceClusters = sourceClustersToRemove => dispatch =>
  dispatch({
    type: REMOVE_SOURCE_CLUSTERS,
    sourceClustersToRemove
  });
