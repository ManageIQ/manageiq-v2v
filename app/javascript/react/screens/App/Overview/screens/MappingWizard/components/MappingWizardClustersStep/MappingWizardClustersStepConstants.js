export const FETCH_V2V_SOURCE_CLUSTERS = 'FETCH_V2V_SOURCE_CLUSTERS';
export const FETCH_V2V_TARGET_CLUSTERS = 'FETCH_V2V_TARGET_CLUSTERS';
export const REMOVE_TARGET_CLUSTER = 'REMOVE_TARGET_CLUSTER';
