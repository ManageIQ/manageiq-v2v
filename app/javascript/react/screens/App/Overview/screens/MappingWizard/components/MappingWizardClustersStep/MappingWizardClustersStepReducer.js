import Immutable from 'seamless-immutable';

import {
  FETCH_V2V_SOURCE_CLUSTERS,
  FETCH_V2V_TARGET_CLUSTERS,
  REMOVE_TARGET_CLUSTER,
  REMOVE_SOURCE_CLUSTERS,
  ADD_TARGET_CLUSTER,
  ADD_SOURCE_CLUSTERS
} from './MappingWizardClustersStepConstants';

const initialState = Immutable({
  sourceClusters: [],
  isFetchingSourceClusters: false,
  isRejectedSourceClusters: false,
  errorSourceClusters: null,
  targetClusters: [],
  isFetchingTargetClusters: false,
  isRejectedTargetClusters: false,
  errorTargetClusters: null
});

export default (state = initialState, action) => {
  switch (action.type) {
    case `${FETCH_V2V_SOURCE_CLUSTERS}_PENDING`:
      return state.set('isFetchingSourceClusters', true);
    case `${FETCH_V2V_SOURCE_CLUSTERS}_FULFILLED`:
      if (
        action.payload.data &&
        action.payload.data.resources &&
        action.payload.data.resources[0]
      ) {
        return state
          .set('sourceClusters', action.payload.data.resources[0].ems_clusters)
          .set('isFetchingSourceClusters', false);
      }
      return state
        .set('sourceClusters', [])
        .set('isFetchingSourceClusters', false);

    case `${FETCH_V2V_SOURCE_CLUSTERS}_REJECTED`:
      return state
        .set('errorSourceClusters', action.payload)
        .set('isRejectedSourceClusters', true)
        .set('isFetchingSourceClusters', false);
    case `${FETCH_V2V_TARGET_CLUSTERS}_PENDING`:
      return state.set('isFetchingTargetClusters', true);
    case `${FETCH_V2V_TARGET_CLUSTERS}_FULFILLED`:
      if (
        action.payload.data &&
        action.payload.data.resources &&
        action.payload.data.resources[0]
      ) {
        return state
          .set('targetClusters', action.payload.data.resources[0].ems_clusters)
          .set('isFetchingTargetClusters', false);
      }
      return state
        .set('targetClusters', [])
        .set('isFetchingTargetClusters', false);
    case `${FETCH_V2V_TARGET_CLUSTERS}_REJECTED`:
      return state
        .set('errorTargetClusters', action.payload)
        .set('isRejectedTargetClusters', true)
        .set('isFetchingTargetClusters', false);
    case REMOVE_TARGET_CLUSTER: {
      return state.set(
        'targetClusters',
        state.targetClusters.filter(
          targetCluster => targetCluster.id !== action.targetClusterToRemove.id
        )
      );
    }
    case REMOVE_SOURCE_CLUSTERS:
      return state.set(
        'sourceClusters',
        state.sourceClusters.filter(
          sourceCluster =>
            !action.sourceClustersToRemove.some(
              clusterToRemove => clusterToRemove.id === sourceCluster.id
            )
        )
      );
    case ADD_TARGET_CLUSTER:
      return state.set('targetClusters', [
        ...state.targetClusters,
        action.targetClusterToAdd
      ]);
    case ADD_SOURCE_CLUSTERS:
      return state.set('sourceClusters', [
        ...state.sourceClusters,
        ...action.sourceClustersToAdd
      ]);
    default:
      return state;
  }
};
