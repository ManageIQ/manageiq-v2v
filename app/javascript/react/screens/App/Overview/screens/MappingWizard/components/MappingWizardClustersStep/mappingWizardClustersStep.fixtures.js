import Immutable from 'seamless-immutable';

export const sourceClusters = Immutable({
  name: 'providers',
  count: 49,
  subcount: 8,
  subquery_count: 8,
  pages: 1,
  resources: [
    {
      href: 'http://localhost:3000/api/providers/10000000000012',
      id: '10000000000012',
      name: 'RHV-M',
      created_on: '2016-07-15T18:25:07Z',
      updated_on: '2017-11-21T23:03:49Z',
      guid: '74cbbbb8-4ab9-11e6-b213-005056b170bd',
      zone_id: '10000000000008',
      type: 'ManageIQ::Providers::Redhat::InfraManager',
      api_version: '3.6.9.2',
      uid_ems: null,
      host_default_vnc_port_start: null,
      host_default_vnc_port_end: null,
      provider_region: null,
      last_refresh_error: null,
      last_refresh_date: '2017-02-26T02:44:22Z',
      provider_id: null,
      realm: null,
      tenant_id: '10000000000001',
      project: null,
      parent_ems_id: null,
      subscription: null,
      last_metrics_error: null,
      last_metrics_update_date: null,
      last_metrics_success_date: null,
      tenant_mapping_enabled: null,
      enabled: true,
      options: {},
      emstype: 'rhevm',
      ems_clusters: [
        {
          href: 'http://localhost:3000/api/clusters/10000000000001',
          id: '10000000000001',
          name: 'VMWare Cluster1',
          ems_id: '10000000000012',
          created_on: '2016-07-15T18:28:16Z',
          updated_on: '2016-07-15T18:28:16Z',
          uid_ems: '00000002-0002-0002-0002-00000000024b',
          ha_enabled: null,
          ha_admit_control: null,
          ha_max_failures: null,
          drs_enabled: null,
          drs_automation_level: null,
          drs_migration_threshold: null,
          last_perf_capture_on: null,
          ems_ref_obj: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          effective_cpu: null,
          effective_memory: null,
          ems_ref: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          type: null
        },
        {
          href: 'http://localhost:3000/api/clusters/10000000000001',
          id: '10000000000002',
          name: 'VMWare Cluster2',
          ems_id: '10000000000012',
          created_on: '2016-07-15T18:28:16Z',
          updated_on: '2016-07-15T18:28:16Z',
          uid_ems: '00000002-0002-0002-0002-00000000024b',
          ha_enabled: null,
          ha_admit_control: null,
          ha_max_failures: null,
          drs_enabled: null,
          drs_automation_level: null,
          drs_migration_threshold: null,
          last_perf_capture_on: null,
          ems_ref_obj: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          effective_cpu: null,
          effective_memory: null,
          ems_ref: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          type: null
        },
        {
          href: 'http://localhost:3000/api/clusters/10000000000001',
          id: '10000000000003',
          name: 'VMWare Cluster3',
          ems_id: '10000000000012',
          created_on: '2016-07-15T18:28:16Z',
          updated_on: '2016-07-15T18:28:16Z',
          uid_ems: '00000002-0002-0002-0002-00000000024b',
          ha_enabled: null,
          ha_admit_control: null,
          ha_max_failures: null,
          drs_enabled: null,
          drs_automation_level: null,
          drs_migration_threshold: null,
          last_perf_capture_on: null,
          ems_ref_obj: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          effective_cpu: null,
          effective_memory: null,
          ems_ref: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          type: null
        },
        {
          href: 'http://localhost:3000/api/clusters/10000000000001',
          id: '10000000000004',
          name: 'VMWare Cluster4',
          ems_id: '10000000000012',
          created_on: '2016-07-15T18:28:16Z',
          updated_on: '2016-07-15T18:28:16Z',
          uid_ems: '00000002-0002-0002-0002-00000000024b',
          ha_enabled: null,
          ha_admit_control: null,
          ha_max_failures: null,
          drs_enabled: null,
          drs_automation_level: null,
          drs_migration_threshold: null,
          last_perf_capture_on: null,
          ems_ref_obj: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          effective_cpu: null,
          effective_memory: null,
          ems_ref: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          type: null
        },
        {
          href: 'http://localhost:3000/api/clusters/10000000000001',
          id: '10000000000005',
          name: 'VMWare Cluster5',
          ems_id: '10000000000012',
          created_on: '2016-07-15T18:28:16Z',
          updated_on: '2016-07-15T18:28:16Z',
          uid_ems: '00000002-0002-0002-0002-00000000024b',
          ha_enabled: null,
          ha_admit_control: null,
          ha_max_failures: null,
          drs_enabled: null,
          drs_automation_level: null,
          drs_migration_threshold: null,
          last_perf_capture_on: null,
          ems_ref_obj: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          effective_cpu: null,
          effective_memory: null,
          ems_ref: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          type: null
        },
        {
          href: 'http://localhost:3000/api/clusters/10000000000001',
          id: '10000000000006',
          name: 'VMWare Cluster6',
          ems_id: '10000000000012',
          created_on: '2016-07-15T18:28:16Z',
          updated_on: '2016-07-15T18:28:16Z',
          uid_ems: '00000002-0002-0002-0002-00000000024b',
          ha_enabled: null,
          ha_admit_control: null,
          ha_max_failures: null,
          drs_enabled: null,
          drs_automation_level: null,
          drs_migration_threshold: null,
          last_perf_capture_on: null,
          ems_ref_obj: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          effective_cpu: null,
          effective_memory: null,
          ems_ref: '/api/clusters/00000002-0002-0002-0002-00000000024b',
          type: null
        }
      ]
    }
  ]
});

export const targetClusters = Immutable({
  name: 'providers',
  count: 49,
  subcount: 1,
  subquery_count: 1,
  pages: 1,
  resources: [
    {
      href: 'http://localhost:3000/api/providers/10000000000001',
      id: '10000000000001',
      name: 'vCenter',
      created_on: '2016-07-08T10:33:40Z',
      updated_on: '2017-02-26T03:18:27Z',
      guid: '700f10d8-44f7-11e6-8d89-005056b170bd',
      zone_id: '10000000000003',
      type: 'ManageIQ::Providers::Vmware::InfraManager',
      api_version: '6.0',
      uid_ems: '0302ea13-8681-4b48-9609-b4aea788ca63',
      host_default_vnc_port_start: null,
      host_default_vnc_port_end: null,
      provider_region: null,
      last_refresh_error: null,
      last_refresh_date: '2017-02-26T03:18:27Z',
      provider_id: null,
      realm: null,
      tenant_id: '10000000000001',
      project: null,
      parent_ems_id: null,
      subscription: null,
      last_metrics_error: null,
      last_metrics_update_date: null,
      last_metrics_success_date: null,
      tenant_mapping_enabled: null,
      enabled: true,
      options: null,
      emstype: 'vmwarews',
      ems_clusters: [
        {
          href: 'http://localhost:3000/api/clusters/10000000000006',
          id: '10000000000007',
          name: 'RHV Cluster1',
          ems_id: '10000000000007',
          created_on: '2016-08-03T02:19:35Z',
          updated_on: '2017-02-26T02:33:43Z',
          uid_ems: 'domain-c7',
          ha_enabled: false,
          ha_admit_control: true,
          ha_max_failures: 1,
          drs_enabled: false,
          drs_automation_level: 'fullyAutomated',
          drs_migration_threshold: 3,
          last_perf_capture_on: null,
          ems_ref_obj: 'domain-c7',
          effective_cpu: 109356,
          effective_memory: 789977235456,
          ems_ref: 'domain-c7',
          type: null
        },
        {
          href: 'http://localhost:3000/api/clusters/10000000000006',
          id: '10000000000008',
          name: 'RHV Cluster2',
          ems_id: '10000000000008',
          created_on: '2016-08-03T02:19:35Z',
          updated_on: '2017-02-26T02:33:43Z',
          uid_ems: 'domain-c7',
          ha_enabled: false,
          ha_admit_control: true,
          ha_max_failures: 1,
          drs_enabled: false,
          drs_automation_level: 'fullyAutomated',
          drs_migration_threshold: 3,
          last_perf_capture_on: null,
          ems_ref_obj: 'domain-c7',
          effective_cpu: 109356,
          effective_memory: 789977235456,
          ems_ref: 'domain-c7',
          type: null
        },
        {
          href: 'http://localhost:3000/api/clusters/10000000000006',
          id: '10000000000009',
          name: 'RHV Cluster3',
          ems_id: '10000000000009',
          created_on: '2016-08-03T02:19:35Z',
          updated_on: '2017-02-26T02:33:43Z',
          uid_ems: 'domain-c7',
          ha_enabled: false,
          ha_admit_control: true,
          ha_max_failures: 1,
          drs_enabled: false,
          drs_automation_level: 'fullyAutomated',
          drs_migration_threshold: 3,
          last_perf_capture_on: null,
          ems_ref_obj: 'domain-c7',
          effective_cpu: 109356,
          effective_memory: 789977235456,
          ems_ref: 'domain-c7',
          type: null
        }
      ]
    }
  ]
});

export const requestSourceClustersData = {
  method: 'GET',
  fetchSourceClustersUrl: '/api/sourceClusters',
  response: { data: sourceClusters }
};

export const requestTargetClustersData = {
  method: 'GET',
  fetchTargetClustersUrl: '/api/targetClusters',
  response: { data: targetClusters }
};

export const initialState = Immutable({
  sourceClusters: [],
  targetClusters: [],
  fetchSourceClustersUrl: requestSourceClustersData.fetchSourceClustersUrl,
  fetchTargetClustersUrl: requestTargetClustersData.fetchTargetClustersUrl
});
