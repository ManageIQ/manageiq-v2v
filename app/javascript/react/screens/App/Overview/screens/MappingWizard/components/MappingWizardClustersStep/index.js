import { connect } from 'react-redux';
import MappingWizardClustersStep from './MappingWizardClustersStep';
import * as MappingWizardClustersStepActions from './MappingWizardClustersStepActions';

import reducer from './MappingWizardClustersStepReducer';

export const reducers = { mappingWizardClustersStep: reducer };

const mapStateToProps = ({ mappingWizardClustersStep }, ownProps) => ({
  ...mappingWizardClustersStep,
  ...ownProps.data
});

const mergeProps = (stateProps, dispatchProps, ownProps) =>
  Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(
  mapStateToProps,
  MappingWizardClustersStepActions,
  mergeProps
)(MappingWizardClustersStep);
