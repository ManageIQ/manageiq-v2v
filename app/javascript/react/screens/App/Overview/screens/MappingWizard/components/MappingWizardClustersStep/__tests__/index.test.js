import React from 'react';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { initialState } from '../mappingWizardClustersStep.fixtures';
import MappingWizardClustersStep from '../MappingWizardClustersStep';
import MappingWizardClustersStepContainer, { reducers } from '../index';

describe('Mapping Wizard integration test', () => {
  const middlewares = [thunk, promiseMiddleware()];
  const generateStore = () =>
    createStore(
      combineReducers({ ...reducers }),
      {
        mappingWizardClustersStep: initialState
      },
      applyMiddleware(...middlewares)
    );

  const mountComponent = store =>
    mount(
      <Provider store={store}>
        <MappingWizardClustersStepContainer />
      </Provider>
    );

  it('should mount the MappingWizardClusterStep with mapStateToProps reduced', () => {
    const store = generateStore();
    const wrapper = mountComponent(store);

    // query the unconnected component to assert reduced props
    const component = wrapper.find(MappingWizardClustersStep);
    expect(component.props()).toMatchSnapshot();

    // isFetching set true after fetch called in `componentDidMount`
    expect(component.props().isFetchingSourceClusters).toBeTruthy();
    expect(component.props().isFetchingTargetClusters).toBeTruthy();
  });
});
