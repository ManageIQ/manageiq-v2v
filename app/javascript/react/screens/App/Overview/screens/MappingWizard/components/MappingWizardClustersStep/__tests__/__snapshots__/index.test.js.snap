// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mapping Wizard integration test should mount the MappingWizardClusterStep with mapStateToProps reduced 1`] = `
Object {
  "fetchSourceClustersAction": [Function],
  "fetchSourceClustersUrl": "/api/sourceClusters",
  "fetchTargetClustersAction": [Function],
  "fetchTargetClustersUrl": "/api/targetClusters",
  "isFetchingSourceClusters": true,
  "isFetchingTargetClusters": true,
  "removeTargetCluster": [Function],
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;
